src/Path.hs

```diff
@@ -370,9 +381,12 @@ filename (Path l) =
 --
 -- The following properties hold:
 --
+-- @dirname $(mkRelDir ".") == $(mkRelDir ".")@
+--
 -- @dirname (p \<\/> a) == dirname a@
 --
 dirname :: Path b Dir -> Path Rel Dir
+dirname (Path []) = Path []
```

Comment:
Not sure what's wrong here? This is just for `dirname "." = "."`, applies only to the `Path Rel Dir` case. `Path []` will never occur for `Path Abs Dir`.